core/trino-main/src/main/java/io/trino/split/SplitSource.java

```diff
@@ -37,6 +38,14 @@ public interface SplitSource
 
     Optional<List<Object>> getTableExecuteSplitsInfo();
 
+    /**
+     * Returns the split source's metrics, mapping a metric ID to its latest value.
+     * Each call must return an immutable snapshot of available metrics.
+     * The same ID metrics are merged across all tasks and exposed via OperatorStats.
+     * This method can be called after the split source is closed.
```

Comment:
This method can be called after the split source is closed and in that case returns last split metrics.